plugins/plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpPostResponse.java

```diff
@@ -128,32 +128,28 @@ private void parseAndSetFormValues(Node samlResponseForm) {
      *
      * And "hidden" means both the SAMLResponse and RelayState MUST be placed in "hidden" form controls
      */
-    // SAMLResponse portion
-    if (samlResponseNode != null) {
 
-      if (isNotEmpty(samlResponseNode.value())) {
-        samlResponse = samlResponseNode.value();
-      } else if (isNotEmpty(samlResponseNode.attributes().get(VALUE))) {
-        samlResponse = samlResponseNode.attributes().get(VALUE);
-      }
-
-      if (isNotEmpty(samlResponseNode.getAttribute(TYPE))) {
-        isSamlResponseHidden = samlResponseNode.getAttribute(TYPE).equals("hidden");
-      }
-    }
-
-    // RelayState portion
-    if (relayStateNode != null) {
+    samlResponse = extractValue(samlResponseNode);
+    isSamlResponseHidden = extractType(samlResponseNode);
 
-      if (isNotEmpty(relayStateNode.value())) {
-        relayState = relayStateNode.value();
-      } else if (isNotEmpty(relayStateNode.attributes().get(VALUE))) {
-        relayState = relayStateNode.attributes().get(VALUE);
-      }
+    relayState = extractValue(relayStateNode);
+    isRelayStateHidden = extractType(relayStateNode);
+  }
 
-      if (isNotEmpty(relayStateNode.getAttribute(TYPE))) {
-        isRelayStateHidden = relayStateNode.getAttribute(TYPE).equals("hidden");
+  private String extractValue(Node node) {
+    if (node != null) {
+      if (isNotEmpty(node.value())) {
+        return node.value();
+      } else if (isNotEmpty(node.attributes().get(VALUE))) {
+        return node.attributes().get(VALUE);
       }
     }
+    return null;
+  }
+
+  private boolean extractType(Node node) {
```

Comment:
✏️ I feel like this name doesn't quite describe its function. Maybe `isTypeHidden()` instead?

Also might be cleaner written as:
```
private boolean isTypeHidden(Node node) {
    return node !=null
        && "hidden".equals(node.getAttribute(TYPE))
}
```

Reply:
I removed that cause it didn't exist when I rebased